api/proto/info/info.proto

```diff
@@ -393,6 +394,37 @@ message ZCellularProvider {
   bool roaming = 4;
 }
 
+// Snapshot states
+enum ZSnapshotState {
+  Z_SNAPSHOT_STATE_UNSPECIFIED = 0;
+  // This state is used when a snapshot is in the process of being
+  // created or an error occurred during the first attempt to create it.
+  // (For example, the operation was delayed)
+  Z_SNAPSHOT_STATE_CREATING = 1;
+  // This state is used when the snapshot has been successfully created.
+  Z_SNAPSHOT_STATE_CREATED = 2;
+  // This state is used when the snapshot is pending deletion or
+  // the first deletion attempt was not successful.
+  Z_SNAPSHOT_STATE_DELETING = 3;
+  // This state is used when the snapshot has been successfully deleted.
+  // Also can get into this state when snapshots were
+  // deleted automatically after the rollback command
+  Z_SNAPSHOT_STATE_DELETED = 4;
+}
+
+// ZInfoSnapshot - Information about snapshot in zfs for zvol
+message ZInfoSnapshot {
+  uint64 creation_time = 1;         // In seconds
+  string uuid = 2;                  // link a command and a response. Is the real name of the snapshot in ZFS
+  string volume_uuid = 3;           // Volume ID of the volume this snapshot belongs to.
+  string display_name = 4;          // Ex: "Tuesday" or creation time (User-friendly name)
+  bool encryption = 5;
+  bool readonly = 6;
```

Comment:
Is readonly about the snapshot or about the volume which was snapshotted? I would think all snapshots are readonly until they become used for a rollback.

Reply:
This property is a snapshot property.

Reply:
OK